src/proxy.py

```diff
@@ -103,13 +103,18 @@ def get_proxies(self, proxies):
     return proxies_list
 
 
-def get_random_proxy(self):
+def get_random_proxy(self, name):
```

Comment:
What type is name? `str`? Could you document that? and add a default value of `None` if that is the intend